### ISSUE TITLE
Fire Allows the Creation of Caves

### ISSUE THREAD

**Issue by RobbyMski** [niels747/2D-Weather-Sandbox#121]
Was working on simulating a volcano when I discovered this. I do think it would be a neat feature to be able to create caves or overhangs without setting the world on fire. Because if you remove the fire (or once the fire burns out) the land will just fall down again.

![Screenshot 2024-03-11 at 09-02-45 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/6d98c54d-7ec5-4f82-86d0-bb4e9790d9b9)

**Comment by niels747** [comment]
I havent been able to recreate this. Did you modify the code to get this?
I might add actual lava in the future

**Comment by RobbyMski** [comment]
I didn't modify any code. But I should have mentioned that I was using the landscaping tool to carve out the shape of the volcano. 

![Screenshot 2024-05-05 at 10-09-44 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/6e20a53e-8474-496c-9a00-72250d1422a8)

![Screenshot 2024-05-05 at 10-11-04 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/84463900-c74e-4f72-8d12-2c4054699699)

As long as there is fire, you can carve out/delete the ground underneath it using the landscaping tool:

![Screenshot 2024-05-05 at 10-10-59 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/30012fac-6999-4ea5-903f-46f6f75ab2d5)

![Screenshot 2024-05-05 at 10-11-04 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/a0205144-9bad-4d4a-8306-25dfcdf0eb5c)

Using this bug: You can also create a floating island. One thing to notice is that the ground underneath the island is also on fire. But when the fire runs out, the ground re-connects with the floating island:

![Screenshot 2024-05-05 at 10-39-13 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/85aac876-82ea-4bf3-a7b4-33d64042fb52)


![Screenshot 2024-05-05 at 10-39-18 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/86f53434-2645-450b-b555-e3508c5009df)


![Screenshot 2024-05-05 at 10-41-03 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/398a03b9-29d3-4104-ac3a-11d7c345d001)

![Screenshot 2024-05-05 at 10-41-08 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/4276bf06-3fc6-470f-a056-fca71d7d74ff)

Recreating this again:
![Screenshot 2024-05-05 at 10-45-45 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/eef3f96d-74e3-4712-978a-e88200f33890)

![Screenshot 2024-05-05 at 10-45-53 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/e7087e83-c6e6-4926-923b-b7414c3e160d)

You can't place land above the fire:


![Screenshot 2024-05-05 at 10-46-14 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/95ac27f1-e0c3-4187-ac4b-6637081aa001)

![Screenshot 2024-05-05 at 10-46-26 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/586a9d38-2417-4b08-871e-f57d384c549b)

Otherwise it will fall through the earth and put it out:

![Screenshot 2024-05-05 at 10-46-52 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/11e3e4ed-16c9-4879-a2ea-b500b5cbe437)

![Screenshot 2024-05-05 at 10-46-48 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/bc0a7693-ea7a-4683-90e9-3a14dac27b58)

But if you place land underneath the burning fire.

![Screenshot 2024-05-05 at 10-47-04 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/1375729a-7fa1-4ad9-9ac6-ed1c321081af)

![Screenshot 2024-05-05 at 10-46-59 2D Weather Sandbox](https://github.com/niels747/2D-Weather-Sandbox/assets/162032759/aa970a3e-af36-4bf9-b5f0-f50dc89bd2db)

And it won't pull the ground on fire above it down.

**Comment by niels747** [comment]
Hi, sorry for the late response. I recreated the bug and it's indeed how you said it: the fireland wasn't programmed to fall down like the normal land does. I fixed it now, but this did inspire me to add actual caves in the future. That will however come with some additional challanges, like IR coming down from the ceiling of the caves and proper boundary conditions in all four directions. You have very good insights for a non programmer! Thanks for reporting the bug!
